### PR TITLE
 [IMP] point_of_sale, pos_sale: add date picker to ship later option

### PR DESCRIPTION
**Pull request by adgu-odoo**
In order to plan a certain Ship Later pos order, a user must go to the backend and find the correct picking linked to the order he made. The date picker can be useful to not have to go back to the backend and continue selling to customers.

Now, when pushing on the button Ship Later, a date picker appears with a default date set as the current date and the user can chose another date after the default date.

Task-id: 3078526

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/point_of_sale/static/src/js/Screens/PaymentScreen/PaymentScreen.js [hunk 1]
```diff
@@ -138,6 +138,38 @@ class PaymentScreen extends PosComponent {
         if (confirmed) {
             this.currentOrder.set_tip(parse.float(payload));
         }
+<<<<<<< HEAD
```
Comment (rhe-odoo) [inline addons/point_of_sale/static/src/js/Screens/PaymentScreen/PaymentScreen.js:141]: Ups

File: addons/point_of_sale/static/src/js/models.js [hunk 2]
```diff
@@ -2613,9 +2613,21 @@ export class Order extends PosModel {
         }
         this.set_partner(partner);
 
+<<<<<<< HEAD
```
Comment (rhe-odoo) [inline addons/point_of_sale/static/src/js/models.js:2616]: ups again

File: addons/point_of_sale/static/src/js/models.js [hunk 3]
```diff
@@ -3595,7 +3623,14 @@ export class Order extends PosModel {
         this.assert_editable();
         this.to_ship = to_ship;
     }
+<<<<<<< HEAD
```
Comment (rhe-odoo) [inline addons/point_of_sale/static/src/js/models.js:3626]: ups

File: addons/point_of_sale/static/src/xml/Popups/DatePickerPopup.xml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+                        <t t-esc="props.confirmText" />
+                    </div>
+                    <div class="button cancel" t-on-click="cancel">
+                        <t t-esc="props.cancelText" />
+                    </div>
+                </div>
+            </div>
+    </t>
+
+</templates>
```
Comment (rhe-odoo) [inline addons/point_of_sale/static/src/xml/Popups/DatePickerPopup.xml:23]: EOF

File: addons/point_of_sale/static/src/js/models.js [hunk 5]
```diff
@@ -2669,6 +2673,16 @@ export class Order extends PosModel {
         }
         return json;
     }
+    export_shipping_date_for_printing() {
```
Comment (trj-odoo) [inline addons/point_of_sale/static/src/js/models.js:2676]: Prefer camelCase again and add `_` in front of the method since you're not using it outside (private)

File: addons/pos_sale/static/src/js/OrderManagementScreen/SaleOrderManagementScreen.js [hunk 6]
```diff
@@ -83,9 +83,10 @@ class SaleOrderManagementScreen extends ControlButtonsMixin(IndependentToOrderSc
             ],
         });
 
-        if (confirmed) {
+        if(confirmed){
```
Comment (trj-odoo) [inline addons/pos_sale/static/src/js/OrderManagementScreen/SaleOrderManagementScreen.js:86]: unnecessary change

File: addons/pos_sale/static/src/js/models.js [hunk 7]
```diff
@@ -33,6 +33,9 @@ const PosSaleOrderline = (Orderline) =>
             this.sale_order_line_id = this.sale_order_line_id || options.sale_order_line_id;
             this.down_payment_details = this.down_payment_details || options.down_payment_details;
             this.customerNote = this.customerNote || options.customer_note;
+            if (this.sale_order_origin_id && this.sale_order_origin_id.shipping_date) {
+                this.order.shipping_date = this.sale_order_origin_id.shipping_date;
```
Comment (trj-odoo) [inline addons/pos_sale/static/src/js/models.js:37]: I'll prefer you calling the method you created to set the shipping date of the order instead of modifying its parents value. That way it's easier to maintain

**Review by trj-odoo — CHANGES_REQUESTED** [review]
Your commit doesn't only impact the `point_of_sale` module, change your title accordingly

**Comment by trj-odoo** [comment]
Also, for the next time, your branch should be "`version`-`tag`-`module(s)`-`feature`-`author`" where `module` can either be super specific if it's about one specific module or you can be general and put "pos" when it's about either point_of_sale or some modules related to it

File: addons/point_of_sale/static/src/js/Screens/PaymentScreen/PaymentScreen.js [hunk 8]
```diff
@@ -138,10 +138,22 @@ class PaymentScreen extends PosComponent {
             this.currentOrder.set_tip(parse.float(payload));
         }
     }
-    toggleIsToShip() {
-        // click_ship
-        this.currentOrder.set_to_ship(!this.currentOrder.is_to_ship());
+    async toggleShippingDatePicker() {
         this.render(true);
```
Comment (trj-odoo) [inline addons/point_of_sale/static/src/js/Screens/PaymentScreen/PaymentScreen.js:142]: useless line

**Review by trj-odoo — APPROVED** [review]

**Comment by trj-odoo** [comment]
@robodoo r+

**Comment by trj-odoo** [comment]
@robodoo r+

**Comment by desdelinux** [comment]
Hi @trj-odoo @adgu-odoo 

Do you plan to backport this to 16.0?

Assuming you're already on 16.1

cc @randall-vx @luisg123v @LuisMzz
